### ISSUE TITLE
UI Request: Create Event Page

### ISSUE THREAD

**Issue by KyleTryon** [TechSquidTV/Shoutify#15]
### Component Type

Full Page

### Component Location

src/pages/app/events.tsx

### What is the component's purpose?

The Events Page will be for creating and sharing multiple types of events across multiple social media channels. Currently, there will be a focus on Twitter and Twitter Spaces, but this page should work to take in any URL resource and share it, or potentially plug into an external system to create the event on the platform of choice automatically.

We will have the ability to schedule Twitter Spaces from within the app right away. It is possible in the future that a YouTube live stream or Twitch live stream could be scheduled in-app, as an example.

![image](https://user-images.githubusercontent.com/33272306/189513923-d9abfc31-d42d-43f7-bd95-ec61c296ad75.png)

## Ideas

- When creating an event, a drop-down could indicate the type of event to create with options such as:
    - Twitter Space
    - YouTube Live Stream
    - URL
- A title and date/time will be needed for scheduling the event
 ---
###UX Consideration: 

- What is the best way to take an event that will be scheduled and turn it into a scheduled or queued post?
    - The user may want to post about the event multiple times before it goes live
    - On Twitter at least, it is common to post a thread where the first tweet talks about the Twitter Space, and maybe has an image, and the second tweet in the thread actually has link to the post

**Comment by Gulshanaggarwal** [comment]
Hey @KyleTryon Twitter doesn't provide API endpoint to create a space. We can only list down spaces. Then creation of Twitter space event isn't possible.

**Comment by KyleTryon** [comment]
Dang, you're right @Gulshanaggarwal. We might be able to make something happen in the future, but I didn't realize the API wouldn't allow scheduling spaces.
https://developer.twitter.com/en/docs/twitter-api/spaces/overview

Maybe we'll need to stick with just putting the URL in there then. Maybe we could potentially even automatically grab the data from the event, but for now maybe we make sure it is input manually?

**Comment by Gulshanaggarwal** [comment]
@KyleTryon The Twitter Space API allows you to only get data rather than a post request so we can't schedule.  The other thing is we can copy space URL from Twitter app but does it make sense to paste it manually in our app? Why someone will do it?

**Comment by KyleTryon** [comment]
> he Twitter Space API allows you to only get data rather than a post request so we can't schedule. The other thing is we can copy space URL from Twitter app but does it make sense to paste it manually in our app? Why someone will

My thought here is that we can help the user create a post sharing their twitter space (or other events, youtube, etc) and we would eventually be able to post them to multiple platforms, not just Twitter.

Some thoughts/ideas:
- This page might feature a Calendar UI to help visualize events timing/planning.
- You may be able to generate calendar reminder links (ICS) for the events with details
  - example: https://parcel.io/tools/calendar
- Maybe events become something that is easy to attach to a new post/tweet from the home page's composer component
- re-share the event before it goes live
- I am not sure what this would look like, but I'm considering events always request an image, and post a thread by default.
  - The first tweet is the description and header image
  - The second tweet is the actual link
  - Having a thread like this allows two embeds, the picture in the first and the link in the second.

**Comment by Cedamar** [comment]
https://user-images.githubusercontent.com/112903111/190878452-013099e9-c0ad-49af-8a1d-699b5aedc192.mov

waiting for feedback

**Comment by Gulshanaggarwal** [comment]
@KyleTryon yes, we can create multiple events where we will have Twitter thread for each event(including images+link) which we can share & also sechdule it for reposting. what do you think?

**Comment by KyleTryon** [comment]
@Cedamar I think we can take advantage of the full right side panel to have a dedicated UI. The links on the left could likely remain simple links to a new page.

@Gulshanaggarwal I am not sure how prominent of a feature it would be to group together events. I think we should focus on publishing for individual events.

**Comment by Gulshanaggarwal** [comment]
@KyleTryon I'm talking about single event. Once user create the event, he can also sechdule like 30 mins or 1 hrs ... prior the event thread again will be sent over the Twitter. What do you think?
